File: MAG/MAG.Model/Donation.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAG.Model
{
    public class Donation
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public decimal Amount { get; set; }

        public DateTime DateDonated { get; set; }

        public string TransactionId { get; set; } = null!;

    }
}

File: MAG/UI/mag_admin/lib/models/donation.dart
import 'package:json_annotation/json_annotation.dart';
part 'donation.g.dart';

@JsonSerializable()
class Donation {
  int? id;
  int? userId;
  double? amount;
  DateTime? dateDonated;
  String? transactionId;

  Donation(
      {this.id,
      this.userId,
      this.amount,
      this.dateDonated,
      this.transactionId});

  factory Donation.fromJson(Map<String, dynamic> json) =>
      _$DonationFromJson(json);

  Map<String, dynamic> toJson() => _$DonationToJson(this);
}
